### DISCUSSION TITLE
Create with same fields does not work

### DISCUSSION THREAD

**Discussion by Laecherlich** [pocketbase/pocketbase#2740] [Q&A] [answered]
I am experiencing a strange issue:

When I want to create a new record in a table using the following:
`    response = await pb.collection(config.database.tables.transaction).create({
      date: '2022-01-01 10:00:00.123Z',
      transaction_type_id: '2ot6xasvg0d0hud',
      comment: 'testasdasdasd',
      transport: true,
    });`

It works the first time and a new record with random id is generated, but when I run this line again nothing happens and result returns the same record as before.

Any ideas why?

**Comment by ganigeorgiev** [comment]
I'm not sure how to reproduce it. Could you provide a more complete code sample showing how and where the above is executed?

**Reply by ganigeorgiev** [reply]
If you are running the above in Next.js 13 it is possible the second response to be a result of the default cache - https://nextjs.org/docs/app/building-your-application/data-fetching/caching#fetch:

> By default, all fetch() requests are cached and deduplicated automatically. **This means that if you make the same request twice, the second request will reuse the result from the first request.**

_(the above link contains several options how to adjust/disable it, for example: `export const fetchCache = 'default-no-store'`)_

**Reply by Laecherlich** [reply]
So I am using Next.js 13.4.6

I have a submit button which triggers a server action:

```
`'use server';

import { transactionType } from '@/config/constants';
import { createTransaction } from './api';

export async function createStorageTransaction({
  date,
  comment,
  transport
}) {
  const transaction = await createTransaction({
    date,
    transaction_type_id: transactionType.storage.id,
    comment,
    transport,
  });
  return transaction;
}`
```

this is the createTransaction function:

```
`export async function createTransaction(data) {
  const pb = await initPocketbase();
  let response;
  try {
    response = await pb
      .collection(config.database.tables.transaction)
      .create(data);
  } catch (error) {
    console.log(error);
  }
  
  return response;
}`
```

and this the initPocketbase function:

```
`export const initPocketbase = async () => {
  try {
    const pb = new PocketBase(config.database.url);
    const authData = await pb.admins.authWithPassword(
      process.env.DATABASE_USER,
      process.env.DATABASE_PASSWORD,
    );
    return pb;
  } catch (error) {
    console.log(error);
  }
};`
```

I will try to adjust the caching.

**Reply by Laecherlich** [reply]
Additional note:

Inside the pocketbase log in the admin ui the second request is not even visible

**Reply by ganigeorgiev** [reply]
> Inside the pocketbase log in the admin ui the second request is not even visible

If that's the case and the `try/catch` doesn't return an error, then most likely is a Next.js cache (under the hood the JS SDK is using `fetch` for sending the request). Please see https://nextjs.org/docs/app/building-your-application/data-fetching/caching#fetch.

**Reply by ganigeorgiev** [reply]
_Side-note:_ If `initPocketbase` is called frequently, it may be a good idea to cache the admin token somewhere to avoid calling `authWithPassword` every time since it could be slow due to the bcrypt hash verification. Once you have an admin token, you can load it with `pb.authStore.save(token, null)`. To refresh the token and the AuthStore state you can call `pb.authStore.isValid && await pb.admins.authRefresh()` (you can call `authWithPassword` when `pb.authStore.isValid` returns false).

**Reply by Laecherlich** [reply]
this sounds good, I just started with this project so iniPocketbase was just my first solution.

**Comment by Laecherlich** [comment]
> > Inside the pocketbase log in the admin ui the second request is not even visible
> 
> If that's the case and the `try/catch` doesn't return an error, then most likely is a Next.js cache (under the hood the JS SDK is using `fetch` for sending the request). Please see https://nextjs.org/docs/app/building-your-application/data-fetching/caching#fetch.

Setting fetchCache unfortunately had no effect

**Reply by ganigeorgiev** [reply]
But where did you add `fetchCache`? Note that there is also a `revalidate` option.

The simplest way to rule out PocketBase issue is to create a static html page in `pb_public/index.html` and open the base url of your PocketBase app:

```html
<!DOCTYPE html />
<html>
    <head>
        <meta charset="utf-8" />
        <title>Response test</title>
    </head>
    <body>
        <script type="module">
            import PocketBase from "https://cdn.jsdelivr.net/gh/pocketbase/js-sdk@v0.15.2/dist/pocketbase.es.mjs"

            const pb = new PocketBase("http://127.0.0.1:8090");

            await pb.admins.authWithPassword("test@example.com", "1234567890");

            const data = {...}

            const record1 = await pb.collection("example").create(data);
            console.log(record1);

            const record2 = await pb.collection("example").create(data);
            console.log(record2);
        </script>
    </body>
</html>
```

**Reply by Laecherlich** [reply, marked as answer]
Adding `revalidatePath('/office/transaction/[type]');` at the bottom of my action fixed it for me :)

**Reply by Laecherlich** [reply]
An even easier fix is to add
`  const cookieStore = cookies();`
at the top of the action. That way nextjs does not deduplicate the post requests.
